pkg/autoscaler/autoscaler.go

```diff
@@ -434,4 +449,72 @@ var _ = Describe("[Feature:Machines] Autoscaler should", func() {
 			framework.WaitForMachineSetDelete(client, ms)
 		}
 	})
+
+	It("It scales from zero", func() {
+
+		// Only run in platforms which support autoscaling from/to zero.
+		clusterInfra, err := framework.GetInfrastructure(client)
+		Expect(err).NotTo(HaveOccurred())
+
+		platform := clusterInfra.Status.PlatformStatus.Type
+		switch platform {
+		case configv1.AWSPlatformType, configv1.GCPPlatformType, configv1.AzurePlatformType:
+			glog.Infof("Platform is %v", platform)
+		default:
+			Skip(fmt.Sprintf("Platform %v does not support autoscaling from/to zero, skipping.", platform))
+		}
+
+		// Anything we create we must cleanup
+		cleanupObjects := map[string]runtime.Object{}
+		defer func() {
+			for name, obj := range cleanupObjects {
+				if err := deleteObject(name, obj); err != nil {
+					glog.Infof("[cleanup] error deleting object %q (%T): %v", name, obj, err)
+				}
+			}
+		}()
+
+		By("Creating a new MachineSet with 0 replicas")
+		machineSetParams := framework.BuildMachineSetParams(client, 0)
+		targetedNodeLabel := fmt.Sprintf("%v-scale-from-zero", autoscalerWorkerNodeRoleLabel)
+		machineSetParams.Labels[targetedNodeLabel] = ""
+
+		machineSet, err := framework.CreateMachineSet(client, machineSetParams)
+		Expect(err).ToNot(HaveOccurred())
+		cleanupObjects[machineSet.GetName()] = machineSet
+
+		framework.WaitForMachineSet(client, machineSet.GetName())
+
+		By(fmt.Sprintf("Creating ClusterAutoscaler "))
+		clusterAutoscaler := clusterAutoscalerResource(100)
+		Expect(client.Create(ctx, clusterAutoscaler)).Should(Succeed())
+		cleanupObjects[clusterAutoscaler.GetName()] = clusterAutoscaler
+
+		expectedReplicas := int32(3)
+		By(fmt.Sprintf("Creating a MachineAutoscaler backed by MachineSet %s/%s - min:%v, max:%v",
+			machineSet.GetNamespace(), machineSet.GetName(), 0, expectedReplicas))
+		asr := machineAutoscalerResource(machineSet, 0, expectedReplicas)
+		Expect(client.Create(ctx, asr)).Should(Succeed())
+		cleanupObjects[asr.GetName()] = asr
+
+		By(fmt.Sprintf("Creating scale-out workload: jobs: %v, memory: %s", expectedReplicas, workloadMemRequest.String()))
+		workload := newWorkLoad(expectedReplicas, workloadMemRequest, targetedNodeLabel)
+		cleanupObjects[workload.GetName()] = workload
+		Expect(client.Create(ctx, workload)).Should(Succeed())
+
+		Eventually(func() bool {
+			ms, err := framework.GetMachineSet(client, machineSet.GetName())
+			Expect(err).ToNot(HaveOccurred())
+
+			By(fmt.Sprintf("Waiting for machineSet replicas to scale out. Current replicas are %v, expected %v.",
```

Comment:
to keep things consistent, could we say "scale up" here?

and also change the other occurrence to "scale down" instead of "scale in"